src/LSST/system/Config.cpp

```diff
@@ -46,45 +46,119 @@ void Config::setup(std::string const& source) {
     _thisPtr = Ptr(new Config(source));
 }
 
-Config::Config(std::string const& source) {
-    if (source == "UNIT_TEST") {
-        // source ignored in this case.
-        _setValuesUnitTests();
-    } else {
-        throw invalid_argument("Config had invalid source " + source);
+void Config::reset() {
+    LCRITICAL("Config reseting global configuration!!!");
+    lock_guard<mutex> lock(_thisMtx);
+    _thisPtr.reset();
+}
+
+Config::Config(std::string const& source) : _source(source) {
+    try {
+        LINFO("Config trying to load yaml file ", _source);
+        _yaml = YAML::LoadFile(_source);
+    } catch (YAML::BadFile& ex) {
+        throw ConfigException(ERR_LOC, string("YAML::BadFile ") + ex.what());
+    }
+    verifyRequiredElements();
+}
+
+void Config::verifyRequiredElements() {
+    try {
+        LINFO("Config::verifyRequiredElements ", _source);
+        string str = getControlServerHost();
+        LINFO(" ControlServer:host=", str);
+
+        int i = getControlServerPort();
+        LINFO("ControlServer:port", i);
+
+        i = getControlServerThreads();
+        LINFO("ControlServer:threads", i);
```

Comment:
Instead of the use of `i`, please consider a more meaningful name such as `port` or `thread`.